CodePush.js

```diff
@@ -170,8 +183,31 @@ function setUpTestDependencies(testSdk, providedTestConfig, testNativeBridge) {
   if (testNativeBridge) NativeCodePush = testNativeBridge;
 }
 
+// This function allows only one syncInternal operation to proceed at any given time.
+// Parallel calls to sync() while one is ongoing yields CodePush.SyncStatus.SYNC_IN_PROGRESS.
+const sync = (() => {
+  let syncInProgress = false;
+  const setSyncCompleted = () => { syncInProgress = false; };
+  
+  return (options = {}, syncStatusChangeCallback, downloadProgressCallback) => {
+    if (syncInProgress) {
+      syncStatusChangeCallback(CodePush.SyncStatus.SYNC_IN_PROGRESS);
+      return Promise.resolve(CodePush.SyncStatus.SYNC_IN_PROGRESS);
+    } 
+    
+    syncInProgress = true;
+    const syncPromise = syncInternal(options, syncStatusChangeCallback, downloadProgressCallback);
+    syncPromise
+      .then(setSyncCompleted)
+      .catch(setSyncCompleted)
+      .done();
```

Comment:
Doesn't calling done here end the Promise chain so that devs couldn't call them on it?